app/components/global_threads/thread_options/index.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {showPermalink} from '@actions/views/permalink';
+import {
+    flagPost,
+    setUnreadPost,
+    unflagPost,
+} from '@mm-redux/actions/posts';
+import {setThreadFollow, updateThreadRead} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getPost} from '@mm-redux/selectors/entities/posts';
+import {getMyPreferences, getTheme} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentTeam, getCurrentTeamUrl} from '@mm-redux/selectors/entities/teams';
+import {getThread} from '@mm-redux/selectors/entities/threads';
+import {isPostFlagged} from '@mm-redux/utils/post_utils';
+import {getDimensions} from '@selectors/device';
+
+import ThreadOptions, {OwnProps} from './thread_options';
+
+import type {GlobalState} from '@mm-redux/types/store';
+
+export function makeMapStateToProps() {
+    return (state: GlobalState, ownProps: OwnProps) => {
+        const myPreferences = getMyPreferences(state);
```

Comment:
this is not a factory, why wouldn't you want to get the most up to date ones ?

Reply:
Apologies .. I didn't understand the question.

Reply:
Why do you have this here?

Reply:
https://github.com/mattermost/mattermost-mobile/blob/a28a3826fabbe01c334cc3d1f176a6f972f99411/app/mm-redux/utils/post_utils.ts#L16 

Because `isPostFlagged` needs preferences.

Reply:
lol, does no really mater as this is now outdated :P